Automattic/newspack-plugin
# Image upload/media modal component

### All Submissions:






### Changes proposed in this Pull Request:



Closes #23 .

This PR introduces a media modal component. The component allows uploading images or selecting existing images using the media modal, and can be pre-populated with an image. I've also added a couple tests for it.

No good pre-existing component exists for us to use, so this is a React wrapper over the standard WP media modal functionality. This should work well for our use.

### How to test the changes in this Pull Request:

1. `npm ci` and `npm run build:webpack`
2. On the Subscriptions page you should see a card with prompt to "Add an image".
3. Test adding and removing images.

#### Discussion questions

- i18n. Should we do it right from the get-go? How should we do it? [Similar to how wc-admin does it](https://github.com/woocommerce/woocommerce-admin/blob/master/package.json#L34)?
- What should it look like once an image has been uploaded? We need to balance being able to see the image with not taking up a ton of space. This is the current design I did (max width of 240px for the image preview):
<img width="561" alt="Screen Shot 2019-05-02 at 11 45 48 AM" src="https://user-images.githubusercontent.com/7317227/57100598-3a63db00-6cd4-11e9-90cc-6614aba69416.png">


### Other information:
